Integrations/TinkoffIntegrationLib/TinkoffBankBroker.cs

```diff
@@ -1,120 +1,77 @@
-        {
-            return GetAllBonds()
-                .Where(b => b.Id == idBond)
-                .Single();
-        }
-
-        /// <summary>
-        /// Method returns list with all bonds from bank broker
-        /// </summary>
-        /// <returns>List with bonds</returns>
-        public List<IMarketInstrument> GetAllBonds()
+        public IEnumerable<Instrument> GetInstruments(string type)
         {
-            return GetInstruments(tinkoffContext.MarketBondsAsync(), tinkoffContext, Depth);
-        }
+            var instruments = new List<Instrument>();
 
-        /// <summary>
-        /// Method returns list with all curencies from bank broker
-        /// </summary>
-        /// <returns>List with currencies</returns>
-        public List<IMarketInstrument> GetAllCurrencies()
-        {
-            return GetInstruments(tinkoffContext.MarketCurrenciesAsync(), tinkoffContext, Depth);
-        }
+            var x = context.MarketBondsAsync().Result;
 
-        /// <summary>
-        /// Method returns specific currency
-        /// </summary>
-        /// <param name="idCurrency">Id required currency, e.g. figi</param>
-        /// <returns>Required currency</returns>
-        public IMarketInstrument GetCurrency(string idCurrency)
-        {
-            return GetAllCurrencies()
-                .Where(c => c.Id == idCurrency)
-                .Single();
-        }
+            var tinkoffInstrumentType = (InstrumentType)Enum.Parse(typeof(InstrumentType), type);
 
-        /// <summary>
-        /// Method returns specific stock
-        /// </summary>
-        /// <param name="idStock">Id required stock, e.g. figi</param>
-        /// <returns>Required stock</returns>
-        public IMarketInstrument GetStock(string idStock)
-        {
-            return GetAllStocks()
-                .Where(st => st.Id == idStock)
-                .Single();
-        }
+            MarketInstrumentList instrumentsList = tinkoffInstrumentType switch
+            {
+                InstrumentType.Bond => context.MarketBondsAsync().Result,
+                InstrumentType.Currency => context.MarketCurrenciesAsync().Result,
+                InstrumentType.Stock => context.MarketStocksAsync().Result,
+                _ => throw new BadRequestException()
+            };
+
+            Parallel.ForEach(instrumentsList.Instruments,
+                (instrument) =>
+                {
+                    try
+                    {
+                        instruments.Add(
+                            new TinkoffInstrumentAdapter(
+                                tinkoffInstrumentType,
+                                instrument,
+                                context.MarketOrderbookAsync(
+                                    instrument.Figi,
+                                    Depth).Result)
+                            );
+                    }
+                    catch (Exception) { }
```

Comment:
Because for some reason Tinkoff API throws weird exceptions here, I have no idea why they do it
There's some instrument's parameter like MinPriceIncrement or sth like that and if it's 0, they throw some exception when we try to get the instrument